pulpcore/cli/rpm/repository.py

```diff
@@ -144,17 +144,32 @@ def repository(ctx: click.Context, pulp_ctx: PulpCLIContext, repo_type: str) ->
     remote_option,
     metadata_signing_service_option,
     click.option(
-        "--metadata-checksum-type", type=click.Choice(CHECKSUM_CHOICES, case_sensitive=False)
+        "--metadata-checksum-type",
+        type=click.Choice(CHECKSUM_CHOICES, case_sensitive=False),
+        help=_("DEPRECATED: Option specifying the checksum type to use for repository metadata."),
     ),
     click.option(
-        "--package-checksum-type", type=click.Choice(CHECKSUM_CHOICES, case_sensitive=False)
+        "--package-checksum-type",
+        type=click.Choice(CHECKSUM_CHOICES, case_sensitive=False),
+        help=_(
+            "DEPRECATED: Option specifying the checksum type to use for packages in "
+            "repository metadata."
+        ),
+    ),
+    pulp_option(
+        "--checksum-type",
+        type=click.Choice(CHECKSUM_CHOICES, case_sensitive=False),
```

Comment:
I have the impression (without digging too deep into pulp_rpm) that this should be limited to the strong checksum algorithms.
Can you add a test to show this option in action?

Reply:
Is the test going to work, if the feature is merged but not yet released?

Reply:
I'd expect one test to "use" the option. Before it's released but merged, it may be guarded with `pulp_rpm>=3.25.0.dev`. The "nighty" test is running with an all sources container.